operation-bench/bench.c

```diff
@@ -0,0 +1,68 @@
+#ifndef WIN32
+#include <sys/time.h>
+#endif
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <math.h>
+#include "tsc_x86.h"
+
+#define N 1e7
+#define CYCLES_REQUIRED 1e7
+#define NUM_RUNS 1
+
+double randf() {
+    return (double)rand() / (double)RAND_MAX;
+}
+
+double* build() {
+    double* x = (double*)malloc(N * sizeof(double));
+    for(size_t i = 0; i < N; ++i) {
+        x[i] = randf();
+    }
+    return x;
+}
+
+#define BENCH_OP(name, expr)                    \
+    num_runs = NUM_RUNS;                        \
+    multiplier = 1.0;                           \
+    do {                                        \
+        num_runs = num_runs * multiplier;       \
+        start = start_tsc();                    \
+        for(i = 0; i < num_runs; ++i) {         \
+            x[i] = expr;                        \
```

Comment:
Is it wanted that we write to the same entry we load from, e.g. for   
```BENCH_OP(addition, x[i] + y[i])```  this will result in ```x[i] = x[i] + y[i]```

Reply:
yes, creating a new array to store results is unnecessary as it storing the results in one of the arguments does not affect computational speed and reduces memory footprint. Therefore it allows to perform smaller measurement before the operation becomes memory bound.